main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -481,6 +483,87 @@ object CompletionProvider {
     }.toList
   }
 
+  /**
+    * Format type params in the right form to be inserted as a snippet
+    */
+  private def formatTParamsSnippet(tparams: List[TypedAst.TypeParam]): String = {
+    tparams match {
+      case Nil => ""
+      case _ => tparams.zipWithIndex.map {
+                  case (tparam, idx) => "$" + s"{${idx + 1}:${tparam.name}}"
+                }.mkString("[", ", ", "]")
+    }
+  }
+
+  /**
+    * Format type params in the right form to be displayed in the list of completions
+    */
+  private def formatTParams(tparams: List[TypedAst.TypeParam]): String = {
+    tparams match {
+      case Nil => ""
+      case _ => tparams.map(_.name).mkString("[", ", ", "]")
+    }
+  }
+
+  val simpleTypeNames = List(
+    "Unit",
+    "Null",
```

Comment:
Null is really specially and should only be used for interop. I would not suggest it as a completion because it would rarely be meaningful.